buildtools/dep/dep.go

```diff
@@ -98,12 +140,23 @@ func UsedIn(dirname string) (bool, error) {
 	return false, nil
 }
 
-// FromFile reads a raw Lockfile from a Gopkg.lock file.
-func FromFile(filename string) (Lockfile, error) {
-	var lockfile Lockfile
-	err := files.ReadTOML(&lockfile, filename)
+// FromFile reads a raw toml Lockfile from a Gopkg.lock file.
+func FromFile(filename string) (Resolver, error) {
```

Comment:
This function name is a bit misleading. Feel free to rearrange and rename functions to make things work better.

Here's what I would do instead:

1. `func New(dir string) (Resolver, error)` that constructs a `Resolver` given a directory containing a `Gopkg.toml` and `Gopkg.lock`.
2. `func NewManifest(filename string) (Manifest, error)` that reads, parses, and constructs a `Manifest` given the path to the `Gopkg.toml`.
3. `func NewLockfile(filename string) (Lockfile, error)` that reads, parses, and constructs a `Lockfile` given the path to the `Gopkg.lock`.